server/src/main/java/org/elasticsearch/action/admin/cluster/repositories/get/TransportGetRepositoriesAction.java

```diff
@@ -80,30 +88,37 @@ protected void masterOperation(
      */
     public static List<RepositoryMetadata> getRepositories(ClusterState state, String[] repoNames) {
         RepositoriesMetadata repositories = state.metadata().custom(RepositoriesMetadata.TYPE, RepositoriesMetadata.EMPTY);
-        if (repoNames.length == 0 || (repoNames.length == 1 && ("_all".equals(repoNames[0]) || "*".equals(repoNames[0])))) {
+        if (isMatchAll(repoNames)) {
             return repositories.repositories();
-        } else {
-            Set<String> repositoriesToGet = new LinkedHashSet<>(); // to keep insertion order
-            for (String repositoryOrPattern : repoNames) {
-                if (Regex.isSimpleMatchPattern(repositoryOrPattern) == false) {
-                    repositoriesToGet.add(repositoryOrPattern);
+        }
+        final List<String> includePatterns = new ArrayList<>();
```

Comment:
I think the new exclude functionality also affects the list repositories API, we should also update the documentation for that API then.

Reply:
++ udpated